### PR TITLE
(feat) ability to extract metadata of an adapter

### PR DESCRIPTION
**Pull request by tphoney**
https://github.com/overmindtech/sdp/issues/264

there are links to 
```
sdp.SubmitSourceHeartbeat
sdp.QueryError
sdp.SourceManaged
sdp.metadata
```
which erroneously still call it source and not adapter

### CODE REVIEW

File: heartbeat.go [hunk 1]
```diff
@@ -43,11 +43,16 @@ func (e *Engine) SendHeartbeat(ctx context.Context) error {
...
 	availableTypesMap := make(map[string]bool)
 	availableScopesMap := make(map[string]bool)
+	adapterMetadata := []*sdp.AdapterMetadata{}
 	for _, source := range e.sh.VisibleSources() {
 		availableTypesMap[source.Type()] = true
 		for _, scope := range source.Scopes() {
 			availableScopesMap[scope] = true
 		}
+		metaData := source.Metadata()
+		if metaData.DescriptiveName != "" {
```
Comment (dylanratcliffe) [inline heartbeat.go:53]: What's this logic here doing? Are we not reporting things that don't have a descriptive name, why?

Comment (tphoney) [reply heartbeat.go]: vestigial code from active development

File: heartbeat_test.go [hunk 2]
```diff
@@ -116,20 +116,6 @@ func TestHeartbeats(t *testing.T) {
...
-		}
-
-		if !slices.Contains(reqAvailableTypes, "test-type") {
-			t.Errorf("expected type 'test-type' to be present in the response")
-		}
-
-		if !slices.Contains(reqAvailableTypes, "test-type2") {
-			t.Errorf("expected type 'test-type2' to be present in the response")
-		}
 	})
```
Comment (tphoney) [inline heartbeat_test.go:119]: we should add the adaptermetadata field to tests.

File: source.go [hunk 3]
```diff
@@ -39,6 +41,29 @@ type Source interface {
 	// return an item for a GET query. In this instance only one item can be
 	// sen on, so the one with the higher weight value will win.
 	Weight() int
+	Metadata() sdp.AdapterMetadata // A struct that contains information about the adapter
```
Comment (dylanratcliffe) [inline source.go:44]: Can we expand on the comment here just to say:

* How metadata is used
* Is it required?
* etc/

Comment (tphoney) [inline source.go:44]: type Source interface becomes type Adapter interface

**Review by dylanratcliffe — APPROVED** [review]

**Pull request merged**